python/ray/data/_internal/planner/filter.py

```diff
@@ -28,5 +32,7 @@ def fn(
             # This causes different behavior between filter and other map-like
             # functions. We should revisit and try to get rid of this logic.
             yield builder.build()
+            if metrics_collector is not None:
```

Comment:
Shall we always make this non None?

Reply:
Sure! Initially this was going to allow the legacy path to not be ported, but we can always collect the metrics and then drop them.